File: training/l.py
n = int(input())

m = {"pink": 1.0}

for x in range(n):
    inp = input().split()
    a = inp[0];
    b = inp[1];
    if not a in m:
        m[a] = 0.0
    if not b in m:
        m[b] = 0.0
    rate = float(inp[2]);
    if (rate * m[b] > m[a]):
        m[a] = rate * m[b]


if not "blue" in m:
    m["blue"] = 0.0
if (m["blue"] > 10.0):
    m["blue"] = 10.0

print(m["blue"])

File: training/l.cpp
#include "bits/stdc++.h"
using namespace std;

#define optimize ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
#define endl "\n"

#define ll long long
#define all(x) x.begin(),x.end()
#define INF 0x3f3f3f3f

int main() {    
    int n;
    cin >> n;

    string a, b;
    double rate;

    unordered_map<string, double> m;
    m["pink"] = 0.0;
    for (int i = 0; i < n; i++) {
        cin >> a >> b >> rate;
        if (m.count(a) == 0) m[a] = -INF;
        if (m.count(b) == 0) m[b] = -INF;
        m[a] = max(m[a], log(rate) + m[b]);
    }

    cout << fixed << setprecision(9) << min(10.0, exp(m["blue"])) << endl;

    return 0;
}